js/privatebin.js

```diff
@@ -4577,6 +4577,11 @@ jQuery.PrivateBin = (function($, RawDeflate) {
          */
         function isInsecureConnection()
         {
+            // use .isSecureContext if available
+            if (window.isSecureContext === true || window.isSecureContext === false) {
+                return !window.isSecureContext;
```

Comment:
As this literally only accepts HTTPS delivered content, I'll remove the whitelist of IPs below again.

On a local environment or for tor etc. a server operator could still disable the warning to be displayed, if it really bothers them. I'll therefore will also skip introducing mechanisms for detecting tor, i2p etc. URLs.

Reply:
It also accepts localhost, actually. And `file://`. And whatever w3c may consider a secure context. (Also .onion in Tor Browser is accepted. It does _exactly_ what we want.)

I.e.
> On a local environment or for tor etc. a server operator could still disable the warning to be displayed, if it really bothers them. I

Not needed.

Please read https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts#When_is_a_context_considered_secure

---

Basically, it was fun that I discovered `window.isSecureContext`, because it actually does what we want (always).
So the idea here is to prefer this (if available, the API may not be there, actually in older browsers) and then fallback to the "old" detection method if not.

Reply:
Reverted d0365faf769ddadd0f1322d54ad3da6323bd3ebf.

Reply:
Ugh, then I'll have to add many more test cases.

Back to the drawing board...